WallsAndHoles/editor.cpp

```diff
@@ -54,9 +59,21 @@ void Editor::createNewMap()
     nmd.exec();
 
     if (nmd.result.width != -1) {
+        if(mTileMap){
+            delete mTileMap;
+        }
         mTileMap = new TileMap(QSize(nmd.result.width, nmd.result.height), this);
         mTileMapToolManager->setTileMap(mTileMap);
 
         mMapView->createMap(mTileMap);
     }
 }
+
+void Editor::heightMap(){
+    mHMap = !mHMap;
+    if(mHMap)
+        mMapView->genHeightMap(mTileMap);
```

Comment:
mTileMap can be null, so if there is no map and this is activated, it will crash